ui/ui-app/src/services/useConfigService.ts

```diff
@@ -136,11 +138,46 @@ export function getRegistryConfig(): ApicurioRegistryConfig {
     return config;
 }
 
-const registryConfig: ApicurioRegistryConfig = getRegistryConfig();
+function overrideObject(base: any, overrides: any | undefined): any {
+    if (overrides === undefined) {
+        return {
+            ...base
+        };
+    }
+    const rval: any = {};
+    Object.getOwnPropertyNames(base).forEach(propertyName => {
+        const baseValue: any = base[propertyName];
+        const overrideValue: any = overrides[propertyName];
+        if (overrideValue) {
+            if (typeof baseValue === "object" && typeof overrideValue === "object") {
+                rval[propertyName] = overrideObject(baseValue, overrideValue);
+            } else {
+                rval[propertyName] = overrideValue;
+            }
+        } else {
+            rval[propertyName] = baseValue;
+        }
+    });
+    return rval;
+}
+
+function overrideConfig(base: ApicurioRegistryConfig, overrides: ApicurioRegistryConfig): ApicurioRegistryConfig {
+    const rval: ApicurioRegistryConfig = cloneObject(base);
+    rval.artifacts = {
+        url: overrides.artifacts.url
+    };
+    rval.ui = overrideObject(rval.ui, overrides.ui);
```

Comment:
I was expecting `overrideObject(rval, overrides);` but, again, no strong objection, just a preference for a different tradeoff.